opus/application/static_media/js/widgets.js

```diff
@@ -135,6 +221,72 @@ var o_widgets = {
         o_widgets.addAttachOrRemoveInputsBehaviors();
     },
 
+    recordRadioButtonStatus: function(slug) {
+        /**
+         * Loop through all radio inputs in a widget and set data-checked to true
+         * if the input is selected, and false if the input is not selected.
+         */
+        $.each($(`#widget__${slug} input[type="radio"]`), function(idx, eachChoice) {
+            if ($(eachChoice).is(":checked")) {
+                $(eachChoice).attr("data-checked", "true");
+            } else {
+                $(eachChoice).attr("data-checked", "false");
+            }
+        });
+    },
+
+    updateSURFACEGEOattrInPlace: function(targetElement, newSurfacegeoTargetSlug, attribute) {
+        /**
+         * Update surfacegeo attributes in place with newly selected target
+         */
+        if (targetElement.length === 0) {
+            return;
+        }
+        let oldTargetStr = `SURFACEGEO${opus.oldSurfacegeoTarget}_`;
+        let newTargetStr = `SURFACEGEO${newSurfacegeoTargetSlug}_`;
+        let newTargetAttr = targetElement.attr(attribute).replace(oldTargetStr, newTargetStr);
+        targetElement.attr(attribute, newTargetAttr);
+    },
+
+    updateSURFACEGEODataObj: function(dataObj, oldTargetStr, newTargetStr) {
+        /**
+         * Update data object (opus.selections, opus.extras, and opus.currentUnitBySlug)
+         * in place with newly selected target.
+         * NOTE: the passed in dataObj will be modified.
+         */
+        for (const oldTargetSlug in dataObj) {
+            if (oldTargetSlug.match(oldTargetStr)) {
```

Comment:
startsWith?

Reply:
For this one, we will stay with .match because it's used to update the dataObj, including opus.extras, and there are "qtype-SURFACEGEO.*" and "unit-SURFACEGEO.*" that have "SURFACEGEO" in the middle of the string. So we will use .match, instead of .startsWith.